### ISSUE TITLE
Access Token SyntaxError

### ISSUE THREAD

**Issue by cyang2-nmdp** [carlowahlstedt/Spark_VSTS_Task#1]
Hello - i'm trying to integrate your Cisco Spark addon into my VSTS build process and am running into this error: 2017-11-14T16:23:31.6873425Z ##[error]SyntaxError: Unexpected token {
It appears this error is due to the Bot Access Token generated. Is there a specific format the token needs to be in? All I'm doing is copying the generated access token and pasting it into the Access Token field within the Spark tool. Could this error be something else besides the Access Token field? I've also attached a screenshot of my error. 

![syntax error](https://user-images.githubusercontent.com/33664899/32792059-0aa6f736-c928-11e7-9527-d8ac4297dc11.jpg)

**Comment by carlowahlstedt** [comment]
I assume you followed the steps to create the token by [creating a Bot](https://developer.ciscospark.com/apps.html)? You should then get an access token for that Bot like this:
<img width="626" alt="screen shot 2017-11-14 at 4 31 34 pm" src="https://user-images.githubusercontent.com/2687687/32806083-65a0b88a-c959-11e7-8be9-44c78e0abe97.png">

If you've done these, you mentioned a VSTS build (as opposed to TFS). What build agent queue are you using?

**Comment by cyang2-nmdp** [comment]
Hi Carlo,

Yes I did. That’s how I got the one time access token. I used that access token to input into the “Access Token” field of the Spark tool. This is after I’ve created my team and space. Once the Team and Space has been created I’ve even added the bot into those team and space. Here’s a screenshot of my bot and the teams as an example. I’ve tried using the regenerated access token several times and got same error. This was the latest Bot.

[cid:image006.jpg@01D35D5F.3E58CE40]

[cid:image007.jpg@01D35D5F.3E58CE40]

[cid:image008.jpg@01D35D5F.3E58CE40]

**Comment by carlowahlstedt** [comment]
You mentioned a VSTS build (as opposed to TFS). What build agent queue are you using?

I guess it's also possible the message your passing to the bot is having issues. What is the message are you attempting to pass?

**Comment by cyang2-nmdp** [comment]
A simple text message "CIT Spark Build Message Test"

**Comment by cyang2-nmdp** [comment]
We use private build agents pool from VSTS. Is VSTS not supported?

**Comment by carlowahlstedt** [comment]
Only other thing that sticks out to me is that it doesn't seem to be able to talk to the API to begin with or otherwise it would output messages when looking for the room to begin with:
<img width="809" alt="screen shot 2017-11-14 at 4 56 18 pm" src="https://user-images.githubusercontent.com/2687687/32807153-f0fee8c2-c95c-11e7-9d8c-dfa34edd3c2f.png">

**Comment by carlowahlstedt** [comment]
I haven't use the private build agent pool in VSTS, but I have used it in host VSTS build queue.

It's possible that your private build agent can't reach the spark servers for some reason?

**Comment by cyang2-nmdp** [comment]
It's possible. I can look into that. It's just the error message threw me off. Was thinking it was more related to the token. Will update after I check the build server. Thanks.

**Comment by cyang2-nmdp** [comment]
I did confirm and my build server can access the Spark team and space so I don't think that's an issue.

**Comment by carlowahlstedt** [comment]
I made some adjustments to the structure of the project to match some others. This was the first one I created and have learned several things since. While doing that, I noticed a compare issue when looking for a team. I've updated the task to fix that compare. I also noticed that I wasn't setting the success/fail result in all cases properly. Hopefully these will correct your issue.

**Comment by cyang2-nmdp** [comment]
Thanks for the update but seems the error is still reoccurring for me. I've tried to reinstall the tool, regenerated new BOT token with no luck. I'm guessing the difference may be hosted builds vs private builds. Maybe i'll wait to see if others run into same issue. Unless there are further updates.

**Comment by cyang2-nmdp** [comment]
Another question, do I need to submit my BOT to the Spark Depot for this to work? This is the only thing I have not done.

**Comment by carlowahlstedt** [comment]
I just checked and the bot I have that's working is not in the Spark Depot, so that doesn't seem to be a requirement.

Did you get the same output that you pasted before but with an updated version number? The version # should be different and I would have expected some different behavior too. I've seem the version get cached from the previously occasionally.

**Comment by cyang2-nmdp** [comment]
Yes I got the same output. I've tried uninstalling and reinstalling the latest version, regenerated new BOT token and still got same error. I've confirmed i'm using the latest version (0.2.7) from my build output. I'm having a different user test it out to see if it's something with my credential.

**Comment by carlowahlstedt** [comment]
Just pushed another update (0.2.9) which adds more logging, more often. Hopefully we can see some output before it just fails. That error just doesn't give much to go off of, but from what I understand that's the proper way to output the error.

Edit: had to fix an issue from previous change that didn't show up in my test build.

**Comment by cyang2-nmdp** [comment]
Great news. Looks like we got past the original error. We realize this is due to our private build agent not being up to date. Once we updated it to latest version we got past the original error. Now we are getting this error (2017-11-21T17:34:11.0990972Z Getting the rooms list
2017-11-21T17:34:11.2551002Z Looking for a match
2017-11-21T17:34:11.2551002Z ##[error]No Room Match Found for that Team)

Are Spaces and Rooms the same thing?
